src/inputs/input/input.jsx

```diff
@@ -264,12 +267,19 @@ class Input extends React.PureComponent {
 
     onKeyDown(event) {
         const {
+            allowDecimals,
             onKeyDown,
         } = this.props;
 
+        const type = this.getType();
+
         if (isFunction(onKeyDown)) {
             onKeyDown(event);
         }
+
+        if (allowDecimals && type === 'number' && event.keyCode === DOT_KEY_CODE) {
+            event.preventDefault();
+        }
```

Comment:
Currently the `<Input>` also allows _letters_ and other non-numeric keys as well!

Wondering if we want to do the following:
* Also add `allowNegativeNumbers` prop as well.
* Basically, if `type === 'number'` we want to _only_ allow (doing `preventDefault()` for anything else):
  * numbers 0 though 9 (always)
  * `.` (if `allowDecimals` is true)
  * `-` (if `allowNegativeNumbers` is true).
  * We might also choose to allow `,` if `allowDecimals` is true.  Obviously there are locale considerations here: Some locales use `,` as the whole number/decimal separator and `.` as the thousands separator (reverse from what EN-US locale does).  But ... one thing at a time!  (We _really_ don't pay a lot of _proper_ attention to l10n/i18n despite what we pretend with that silly thing in the UI solution).

**_Not a Blocker_** for this PR, but seems to me it might be nice to take care of this more comprehensively, while we "have the hood up", so to speak.  Up to you! =)

Reply:
Resolving.  Latest version as of 8429b8f is what we're looking for IMO.